backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/TransferDiskImageCommand.java

```diff
@@ -1478,10 +1510,11 @@ protected void endSuccessfully() {
     }
 
     @Override
-    protected void endWithFailure() {
+    protected void  endWithFailure() {
         log.error("Failed to transfer disk '{}' for image transfer '{}'", getParameters().getImageId(), getCommandId());
         if (getParameters().getTransferType() == TransferType.Upload) {
             // Do rollback only for upload - i.e. remove the disk added in 'createImage()'
+            setImageStatus(ImageStatus.OK);
```

Comment:
My initial version went straight to endWithFailure, so the disk was locked, but now it should go to handleFinalizingFailure first which should take care of it, so it can probably be removed

Reply:
there is actually no reason not to go directly to endWithFailure if no host was found so it's ok, I'll add some more information in a comment

But this PR needs some more work to handle a missing ticket for failures that come after